fv3core/decorators.py

```diff
@@ -247,8 +248,18 @@ def __call__(self, *args, origin: Index3D, domain: Index3D, **kwargs) -> None:
                 **self.backend_kwargs,
             }
 
-            # gtscript.stencil always returns a new class instance even if it
-            # used the cached module.
+            # we ensure sequential compilation to reduce memory pressure
```

Comment:
Just a side note. This is a rather unintuitive way to implement ordering using MPI send and recv. The "standard" way of doing this is by doing...
```
for i = 1, number_of_ranks
    if (my_ranks == i) {
      # do work
    }
    MPI_Barrier()
```